bin/install-pr

```diff
@@ -9,63 +9,193 @@
+      exit 1
+      ;;
+    *)
+      break
+  esac
+  shift
+done
+
 pr=$1
 
 if [ -z "$pr" ]
 then
-  echo "usage: ${0##*/} (PR number)" >&2
+  echo "Usage: ${0##*/} (PR number)" >&2
   exit 1
 fi
 
 if [ -z "$GITHUB_TOKEN" ]
 then
   # shellcheck disable=SC2016
-  echo 'Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
+  echo 'Error: Generate a personal access token at https://github.com/settings/tokens and set it in the $GITHUB_TOKEN env var'
   exit 1
 fi
 
+# Get the URL for downloading the artifacts archive.
 auth="Authorization: token $GITHUB_TOKEN"
+branch=$(curl -sL "$linkerd2_pulls_url/$pr" | jq -r '.head.ref')
+artifacts=$(curl -sL "$linkerd2_kind_integration_url/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
+archive=$(curl -sL -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
 
-branch=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/pulls/$pr" | jq -r '.head.ref')
-
-artifacts=$(curl -s -H "$auth" "https://api.github.com/repos/linkerd/linkerd2/actions/workflows/kind_integration.yml/runs?branch=$branch" | jq -r '.workflow_runs[0].artifacts_url')
-
-archive_url=$(curl -s -H "$auth" "$artifacts" | jq -r '.artifacts[0].archive_download_url')
-
-dir=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
+bindir=$( cd "${BASH_SOURCE[0]%/*}" && pwd )
+rootdir=$( cd "$bindir"/.. && pwd )
+archive_tmp=$(mktemp -d -t "linkerd-pr-$pr.XXXXXXXXXX")
 
-cd "$dir" || exit
-
-echo "### Downloading images ###"
-
-curl -L -o archive.zip -H "$auth" "$archive_url"
+# Change directory temporarily and download artifact.
+cd "$archive_tmp"
 
+curl -L -o archive.zip -H "$auth" "$archive"
 unzip -o archive.zip
 
-echo "### Pushing images ###"
-
-for archive in cli-bin cni-plugin controller debug grafana proxy web
-do 
-  image=$(docker load -i "$archive.tar" | sed "s/Loaded image: //")
-  docker push "$image"
-  tag=$(echo "$image" | cut -f 2 -d ":")
-done
+# Install the Linkerd CLI.
+image=$(docker load -i cli-bin.tar | sed 's/Loaded image: //')
+container_id=$(docker create "$image")
+
+case $(uname) in
+  Darwin)
+    platform="darwin"
+    ;;
+  Linux)
+    platform="linux"
+    ;;
+  *)
+    platform="windows"
+    ;;
+esac
+
+docker cp "$container_id":/out/linkerd-"$platform" "$rootdir"/target/cli/"$platform"/linkerd
+installed_cli=1
+
+# Load images into local docker images
+if [ $is_kind ]
+then
+  # Create new KinD cluster
+  cluster_name="pr-$pr"
+  context_flag=(--context kind-"$cluster_name")
+
+  if "$bindir"/kind get clusters | grep "$cluster_name"
+  then
+    echo "Error: KinD cluster '$cluster_name' exists"
+    exit 1
+  fi
+  
+  "$bindir"/kind create cluster --name "$cluster_name"
+
+  # Load images into cluster
+  for image in cni-plugin controller debug grafana proxy web
+  do
+    echo "Loading $image image into cluster.."
+    "$bindir"/kind load image-archive --name "$cluster_name" $image.tar || tee load_fail &
```

Comment:
how long does it take to load each of these images into kind?  this parallelism machinery is really scary to me.

Reply:
I took this from [kind_integration.yml](https://github.com/linkerd/linkerd2/blob/88cafa36c6683e59a22c3cb4aa3cd5eda6c9a0fe/.github/workflows/kind_integration.yml#L144-L145) where we do the same thing.

I think when I added that it saved about a minute in CI. I think it's valuable for quickly getting an install up and running.